.github/workflows/build.yml

```diff
@@ -0,0 +1,96 @@
+name: Build
+
+on:
+  push: 
+    branches: [staging, master]
+  pull_request:
+    branches: [staging, master]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 8
+      
+      - name: Install Dependencies
+        run: yarn install --immutable --immutable-cache
+      
+      - name: Lint Check
+        run: yarn lint
+
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 8
+      
+      - name: Install Dependencies
+        run: yarn install --immutable --immutable-cache
+      
+      - name: Lint Check
+        run: yarn test
+  
+  codecov:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 8
+      
+      - name: Install Dependencies
+        run: yarn install --immutable --immutable-cache
+      
+      - name: Build App
+        run: yarn run build 
+      
+      - name: Upload Codecov Reports
+        uses: codecov/codecov-action@v2
+    
+  
+  deploy:
+    runs-on: ubuntu-latest
+    needs: [lint, test, codecov]
+    if: github.event_name == 'push'
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 8
+      
+      - name: Install Dependencies
+        run: yarn install --immutable --immutable-cache
+      
+      - name: Build App
+        run: yarn run build
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        
+      - name: Deploy to S3 | Staging
+        if: github.ref == 'refs/heads/staging'
+        run: aws s3 sync ./public/ s3://cru-givestage --delete
+
+      - name: Deploy to S3 | Production
+        if: github.ref == 'refs/heads/master'
+        run: aws s3 sync ./public/ s3://cru-giveprod --delete
```

Comment:
I think `--delete` could be somewhat risky on production. We don't want to delete files before new ones are deployed or have the deploy fail halfway or something. https://github.com/aws/aws-cli/issues/1417

Keeping it on staging is fine... I always thought we should find a better process to cleanup old files on S3 but there's no automated way currently. We could take the suggestion from that issue and run the sync then the sync again right after with `--delete`. Or we could just migrate to AWS Amplify or something where we don't have to worry about this implementation detail.